packages/valory/skills/mech_interact_abci/behaviours/request.py

```diff
@@ -328,6 +411,86 @@ def _build_unwrap_tokens_tx(self) -> WaitableConditionType:
+        status = yield from self.contract_interact(
+            performative=ContractApiMessage.Performative.GET_RAW_TRANSACTION,  # type: ignore
+            contract_address=self.params.nvm_balance_tracker_address,
+            contract_public_id=self.nvm_balance_tracker_contract_id,
+            contract_callable=contract_callable,
+            data_key=data_key,
+            placeholder=placeholder,
+            **kwargs,
+        )
+        return status
+
+    def get_nvm_balance(self):
+        """Get the NVM balance."""
+        status = self._nvm_balance_tracker_contract_interact(
+            contract_callable="get_balance",
+            data_key="balance",
+            placeholder="_nvm_balance",
+            address=self.synchronized_data.safe_contract_address,
+        )
+        return status
+
+    def get_subscription_nft(self):
+        """Get the subscription NFT."""
+        status = self._nvm_balance_tracker_contract_interact(
+            contract_callable="get_subscription_nft",
+            data_key="address",
+            placeholder="_olas_subscription_address",
+        )
+        return status
+
+    def get_subscription_token_id(self):
+        """Get the subscription NFT."""
+        status = self._nvm_balance_tracker_contract_interact(
+            contract_callable="get_subscription_token_id",
+            data_key="id",
+            placeholder="_subscription_id",
+        )
+        return status
+
+    def _olas_subscription_contract_interact(
+        self, contract_callable: str, data_key: str, placeholder: str, **kwargs: Any
+    ) -> WaitableConditionType:
+        """Interact with the OLAS subscription contract."""
+        status = yield from self.contract_interact(
+            performative=ContractApiMessage.Performative.GET_RAW_TRANSACTION,  # type: ignore
+            contract_address=self.olas_subscription_address,
+            contract_public_id=IERC1155.contract_id,
+            contract_callable=contract_callable,
+            data_key=data_key,
+            placeholder=placeholder,
+            **kwargs,
+        )
+        return status
+
+    def get_olas_subscription_balance(self):
+        """Get the OLAS subscription's balance."""
+        status = self._olas_subscription_contract_interact(
+            contract_callable="get_balance",
+            data_key="balance",
+            placeholder="_olas_subscription_balance",
+            account=self.synchronized_data.safe_contract_address,
+            subscription_id=self.subscription_id,
+        )
+        return status
+
+    def set_total_nvm_balance(self) -> Generator:
+        """Get teh total NVM balance."""
+        steps = [
+            self.get_nvm_balance,
+            self.get_subscription_nft,
+            self.get_subscription_token_id,
+            self.get_olas_subscription_balance,
+        ]
+        for step in steps:
+            yield from self.wait_for_condition_with_sleep(step)
```

Comment:
Same with the above, it is a bit confusing why we mix everything.

Reply:
https://github.com/valory-xyz/mech-interact/pull/2#discussion_r2300535554